### PR TITLE
Fix Add topic

### PR DESCRIPTION
**Pull request by 1mr124**

### CODE REVIEW

File: FlaskSite/api/v1/topic.py [hunk 1]
```diff
@@ -5,22 +5,26 @@
 bp = Blueprint('topic', __name__)
 
 
-@bp.route('/create', methods=['POST'])
+@bp.route('/topic/create', methods=['POST'])
```
Comment (MohamedSamehMohamed) [inline FlaskSite/api/v1/topic.py:8]: can we make a global prefix topic? 
to not have to write topic in each api
can we also make the create api only a post method for topic, ...api/v1/topic [post]
check that 
https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5

File: FlaskSite/api/v1/topic.py [hunk 2]
```diff
@@ -5,22 +5,26 @@
...
-            return jsonify({"message": "Topic created successfully"}), 201
-        except Exception as e:
-            db.session.rollback()
-            return jsonify({"message": "An error occurred"}), 500
-    else:
-        return jsonify({"errors": form.errors}), 400
+    data = request.get_json()
+
+    # Validate data
+    if not data or 'title' not in data or 'parent' not in data:
```
Comment (MohamedSamehMohamed) [inline FlaskSite/api/v1/topic.py:13]: make a list of items you want to find in the request, and check if any item of the list not present in the request, 
for readability and easy to modify later

File: FlaskSite/api/v1/topic.py (hunk 1 above)
Comment (MohamedSamehMohamed) [reply FlaskSite/api/v1/topic.py:8]: hint: remove create word

Comment (1mr124) [reply FlaskSite/api/v1/topic.py:8]: i was just testing the ui

**Pull request merged**